core/src/main/java/com/linecorp/armeria/server/cors/CorsPolicy.java

```diff
@@ -0,0 +1,300 @@
+     *
+     * <p>When making a preflight request the client has to perform two request with can be inefficient.
+     * This setting will set the CORS {@code "Access-Control-Max-Age"} response header and enables the
+     * caching of the preflight response for the specified time. During this time no preflight
+     * request will be made.
+     *
+     * @return the time in seconds that a preflight request may be cached.
+     */
+    public long maxAge() {
+        return maxAge;
+    }
+
+    /**
+     * Returns a set of headers to be exposed to calling clients.
+     *
+     * <p>During a simple CORS request only certain response headers are made available by the
+     * browser, for example using:
+     * <pre>{@code
+     * xhr.getResponseHeader("Content-Type");
+     * }</pre>
+     * The headers that are available by default are:
+     * <ul>
+     *   <li>{@code Cahce-Control}</li>
+     *   <li>{@code Content-Language}</li>
+     *   <li>{@code Content-Type}</li>
+     *   <li>{@code Expires}</li>
+     *   <li>{@code Last-Modified}</li>
+     *   <li>{@code Pragma}</li>
+     * </ul>
+     *
+     * <p>To expose other headers they need to be specified, which is what this method enables by
+     * adding the headers names to the CORS {@code "Access-Control-Expose-Headers"} response header.
+     *
+     * @return the list of the headers to expose.
+     */
+    public Set<AsciiString> exposedHeaders() {
+        return exposedHeaders;
+    }
+
+    /**
+     * Returns the allowed set of request methods. The Http methods that should be returned in the
+     * CORS {@code "Access-Control-Request-Method"} response header.
+     *
+     * @return the {@link HttpMethod}s that represent the allowed request methods.
+     */
+    public Set<HttpMethod> allowedRequestMethods() {
+        return allowedRequestMethods;
+    }
+
+    /**
+     * Returns the allowed set of Request Headers.
+     *
+     * <p>The header names returned from this method will be used to set the CORS
+     * {@code "Access-Control-Allow-Headers"} response header.
+     */
+    public Set<AsciiString> allowedRequestHeaders() {
+        return allowedRequestHeaders;
+    }
+
+    /**
+     * Determines if the policy allows a null origin.
+     */
+    public boolean isNullOriginAllowed() {
+        return nullOriginAllowed;
+    }
+
+    /**
+     * Returns HTTP response headers that should be added to a CORS preflight response.
+     *
+     * @return {@link HttpHeaders} the HTTP response headers to be added.
+     */
+    public HttpHeaders preflightResponseHeaders() {
+        return preflightResponseHeaders;
+    }
+
+    /**
+     * This is a non CORS specification feature which enables the setting of preflight
+     * response headers that might be required by intermediaries.
+     *
+     * @param headers the {@link HttpHeaders} to which the preflight headers should be added.
+     */
+    void setPreflightHeaders(final HttpHeaders headers) {
```

Comment:
Perhaps we can remove this method and let the caller do the following?

```java
headers.add(policy.generatePreflightResponseHeaders());
```

Reply:
I think this method would be better remaining for consistency because all of the other fields (such as allow methods, allow headers etc) do have the setters for the headers. What do you think?

Reply:
Sounds reasonable to me. It's fine to keep this setter. :-)